contribs/github-bot/comment.go

```diff
@@ -0,0 +1,276 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/gnolang/gno/contribs/github-bot/client"
+
+	"github.com/google/go-github/v64/github"
+	"github.com/sethvargo/go-githubactions"
+)
+
+// These structures contain the necessary information to generate
+// the bot's comment from the template file
+type AutoContent struct {
+	Description        string
+	Satisfied          bool
+	ConditionDetails   string
+	RequirementDetails string
+}
+type ManualContent struct {
+	Description      string
+	ConditionDetails string
+	CheckedBy        string
+	Teams            []string
+}
+type CommentContent struct {
+	AutoRules    []AutoContent
+	ManualRules  []ManualContent
+	allSatisfied bool
+}
+
+// getCommentManualChecks parses the bot comment to get the checkbox status,
+// the check description and the username who checked it
+func getCommentManualChecks(commentBody string) map[string][2]string {
+	checks := make(map[string][2]string)
+
+	reg := regexp.MustCompile(`(?m:^- \[([ x])\] (.+)?$)`)
+	subReg := regexp.MustCompile(`(?m:(.+) \(checked by @(\w+)\)$)`)
+	matches := reg.FindAllStringSubmatch(commentBody, -1)
+
+	for _, match := range matches {
+		if subMatches := subReg.FindAllStringSubmatch(match[2], -1); len(subMatches) > 0 {
+			checks[subMatches[0][1]] = [2]string{match[1], subMatches[0][2]}
+		} else {
+			checks[match[2]] = [2]string{match[1]}
+		}
+	}
+
+	return checks
+}
+
+// handleCommentUpdate checks if:
+//   - the current run was triggered by GitHub Actions
+//   - the triggering event is an edit of the bot comment
+//   - the comment was not edited by the bot itself (prevent infinite loop)
+//   - the comment change is only a checkbox being checked or unckecked (or restore it)
+//   - the actor / comment editor has permission to modify this checkbox (or restore it)
+func handleCommentUpdate(gh *client.GitHub) error {
+	// Get GitHub Actions context to retrieve comment update
+	actionCtx, err := githubactions.Context()
+	if err != nil {
+		gh.Logger.Debugf("Unable to retrieve GitHub Actions context: %v", err)
+		return nil
+	}
+
+	// Ignore if it's not a comment related event
+	if actionCtx.EventName != "issue_comment" {
+		gh.Logger.Debugf("Event is not issue comment related (%s)", actionCtx.EventName)
+		return nil
+	}
+
+	// Ignore if the action type is not deleted or edited
+	actionType, ok := actionCtx.Event["action"].(string)
+	if !ok {
+		return fmt.Errorf("unable to get type on issue comment event")
+	}
+
+	if actionType != "deleted" && actionType != "edited" {
+		return nil
+	}
+
+	// Return if comment was edited by bot (current authenticated user)
+	authUser, _, err := gh.Client.Users.Get(gh.Ctx, "")
+	if err != nil {
+		return fmt.Errorf("unable to get authenticated user: %w", err)
+	}
+
+	if actionCtx.Actor == authUser.GetLogin() {
+		gh.Logger.Debugf("Prevent infinite loop if the bot comment was edited by the bot itself")
+		os.Exit(0)
```

Comment:
This function originally did not return any error and used `os.Exit(1)` in case of an error or `os.Exit(0)` in this specific case, but I changed that because it was a required change.

I will use a standard error instead, but knowing that this function is private and will only ever be called by the main function (which will check if this error is standard and return Exit(0)).